### DISCUSSION TITLE
Semantic Tree / Conceptual Hierarchy Syntax Ideas?

### DISCUSSION THREAD

**Discussion by manunamz** [svsool/vscode-memo#518] [Ideas]
Hi all,

I've become a fan of hierarchical tags, conceptual hierarchies, and tree structures in general to help sort notes. I want to take this concept a bit further with these ideas in mind: 

> "It is important to view knowledge as sort of a semantic tree — make sure you understand the fundamental principles, ie the trunk and big branches, before you get into the leaves/details or there is nothing for them to hang on to."
> 
> ~ [Elon Musk][quote-musk]; (my interpretation [here][manunam-musk])

...and:

> "It is true that it is primarily factual information that [one] wants...But [one] should not be content with facts in isolation. The encyclopedia presents [us] with an arrangement of facts—facts in relation to other facts. The understanding, as contrasted with the mere information, that an encylopedia can provide depends on the recognition of such relations.
> 
> In an alphabetically-arranged encyclopedia, these relations are to a large extent obscured. In a topically-arranged encyclopedia, they are, of course, highlighted."
> 
> ~ [How To Read A Book][amzn-book]

So, I'd like to come up with a syntax that facilitates the creation, management, and viewing of "**semantic trees**" or "**topically arranged encyclopedias**". Imagine pushing hierarchical tagging to encompass all the words you use.

It would be neat to see `vscode-memo` support these ideas. I would simply make a feature request, but I'm also not 100% sure what the precise syntax and behavior should be. I wrote a [long blerb][gist] on the different syntaxes I've been mulling over and I have a working example [here][jekyll-namespaces][^impl], which then is put into action [here][jekyll-bonsai].

_But I was wondering what others thoughts and experiences were with different syntaxes that help with hierarchy management?_

---

Note: It is just now occurring to me that a combination of `vscode-memo`'s pre-existing "filename clash" handling (e.g. file paths in links) and forward slashes in filenames might be a powerful combination here...

[^impl]: Essentially, I'm using `dot.name.spacing.md` and building a tree based solely on these filenames ([inspired by...][dendron]). Major strengths of this syntax include being able to distinguish between terms (ex: "tree" is easy to distinguish when its conceptual lineage is preserved: `comp-sci.graph.tree` vs `botany.plant.tree`) and being relatively easy to manipulate via regular expressions.

[amzn-book]: <https://www.amazon.com/How-Read-Book-Classic-Intelligent/dp/0671212095>
[dendron]: <https://wiki.dendron.so/notes/f3a41725-c5e5-4851-a6ed-5f541054d409/>
[gist]: <https://gist.github.com/manunamz/7b713cd9e65e72d7d99c1b77cb2f7a11>
[jekyll-bonsai]: <https://jekyll-bonsai.netlify.app/entry/fa6aa0c597/>
[jekyll-namespaces]: <https://www.github.com/manunamz/jekyll-namespaces/>
[manunam-musk]: <https://www.manunam.me/post/a685b1968d/>
[quote-musk]: <https://www.reddit.com/r/IAmA/comments/2rgsan/i_am_elon_musk_ceocto_of_a_rocket_company_ama/cnfre0a/?utm_source=share&utm_medium=web2x&context=3>

**Comment by svsool** [comment]
Hi, @manunamz!

I read some of your notes and the approach you arrived at which is using namespaces. I won't pretend to know the ideal solution for the topic in question either.

In Memo similar thing to namespaces would probably be achieved by using short references and placing equally named files under different folders. In case of collisions Memo will fallback to longer reference on autocomplete.

My knowledge base is pretty simple and relatively flat. I arrange some notes under folders when I see connections otherwise I use it for creating notes quickly on various topics without giving too much attention to hierarchy.

**Comment by manunamz** [comment]
Thanks for giving the brain-dump a look-over. 

> "In Memo similar thing to namespaces would probably be achieved by using short references and placing equally named files under different folders."

I will definitely play around with this.

> "...I arrange some notes under folders when I see connections otherwise I use it for creating notes quickly on various topics without giving too much attention to hierarchy."

If a vault is just side notes, daily notes, todo lists, and generally content that is not meant to accumulate long-term, a hierarchy like this could be overkill. It's mostly useful once you reach so many notes navigation becomes a headache. The only reason I don't just default to directories is because of the duplication that would be required to represent a note that is simultaneously a directory (as mentioned in the [blurb](https://gist.github.com/manunamz/7b713cd9e65e72d7d99c1b77cb2f7a11)).

As a side note, this was another [conversation](https://discuss.logseq.com/t/proposal-changing-how-namespaces-function-in-logseq/3727) about namespacing mechanics and syntax.
